app/models/user.rb

```diff
@@ -15,7 +15,7 @@ class User < ActiveRecord::Base
   has_many :clients, through: :company
 
   ## Constants
-  ROLES = ["auditor", "accountant", "manager"]
+  enum role: ["auditor", "accountant", "manager", "admin"]
 
   ## Callbacks
   after_create :assign_role_and_admin_id
```

Comment:
There is only one user that can be admin per company and if this method makes an user admin then we must not use callback as it will be invoked for each type of user.

Instead call this method manually from controller.